### ISSUE TITLE
Support yarnv2 (berry)

### ISSUE THREAD

**Issue by searchableguy** [pubkey/rxdb#2098]
### Description
Currently, you can't install rxdb with yarn berry.

### Reproducible steps
1. In a directory, run `yarn init -2` 
2. Run `yarn add rxdb`

**Comment by pubkey** [comment]
What is the error?
You are right we should put install via yarn to the CI.

**Comment by searchableguy** [comment]
https://github.com/yarnpkg/berry/issues/885

**Comment by pubkey** [comment]
I have no idea how to fix this. Can you help me out?

**Comment by searchableguy** [comment]
Ok I will look into fixing this and report.

**Comment by trik** [comment]
the bug it's not directly related to rxdb, but to its dependencies
package es3ify@0.1.3 declares esprima dependency like this
[`"esprima-fb": "~3001.0001.0000-dev-harmony-fb"`](https://github.com/sophiebits/es3ify/blob/de5095700270fe8a4d5761c107925b5e50ec8d6a/package.json#L7)
the problem does not occur with newer versions of es3ify
es3ify is a transitive dependency coming from pouchdb-all-dbs which is a poorly maintained plugin, last commit is from 2016.. we should consider removing this dependency also cause the plugin is added in [plugins/server.ts](https://github.com/pubkey/rxdb/blob/90cf0367ebdf0d9d3cc63fef5286393e8466ee64/src/plugins/server.ts#L36) but the functions exposed by the plugin (allDbs and resetAllDbs) are never used

**Comment by pubkey** [comment]
I tried to fix this at ef2e9e6feb12e605d251661cce1ddb0046a8302c but I give up. Yarn is painful to use. It is creating strange folders even in the root of the project.

I removed `pouchdb-all-db` from the dependencies but that did not help. Please help me out.

**Comment by trik** [comment]
@pubkey as a workaround, you can force es3ify version to a newer one in package.json

```
{
  "name": "test",
  "dependencies": {
    "rxdb": "^9.0.0"
  },
  "resolutions": {
    "es3ify": "0.2.2"
  }
}
```

**Comment by pubkey** [comment]
I am closing this. A PR for the fix is still welcomed. WIP on [this branch](https://github.com/pubkey/rxdb/tree/yarn2)

**Comment by liyachun01** [comment]
> I tried to fix this at [ef2e9e6](https://github.com/pubkey/rxdb/commit/ef2e9e6feb12e605d251661cce1ddb0046a8302c) but I give up. Yarn is painful to use. It is creating strange folders even in the root of the project.
> 
> I removed `pouchdb-all-dbs` from the dependencies but that did not help. Please help me out.

Yarn v2 has its own [answer](https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored) on if you should ignore those strange folders or not, FYI.
